### PR TITLE
Allow controlling of processing of frames by RemoteControlClient

### PR DESCRIPTION
**Pull request by mrlacey**
## PR Type

What kind of change does this PR introduce?

- other: Remote Control behavior extensibility

## What is the current behavior?

All received frames are processed by the RemoteControlClient

## What is the new behavior?

It's now possible to register a class to "pre-process" all frames received by the RemoteControlClient. This can do its own work with a received frame and, optionally, prevent the default processing of the frame too.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.csproj [hunk 1]
```diff
@@ -43,6 +43,7 @@
 	  <UpToDateCheckInput Remove="HotReload\Messages\IMessage.cs" />
 	  <UpToDateCheckInput Remove="HotReload\Messages\XamlLoadError.cs" />
 	  <UpToDateCheckInput Remove="IRemoteControlProcessor.cs" />
+	  <UpToDateCheckInput Remove="IRemoteControlPreProcessor.cs" />
 	  <UpToDateCheckInput Remove="ProjectConfigurationAttribute.cs" />
 	  <UpToDateCheckInput Remove="RemoteControlClient.cs" />
```
Comment (Youssef1313) [inline src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.csproj:48]: Why do we have all these? Doesn't look right to me

File: src/Uno.UI.RemoteControl/RemoteControlClient.cs [hunk 2]
```diff
@@ -80,6 +81,11 @@ private void RegisterProcessor(IRemoteControlProcessor processor)
 			_processors[processor.Scope] = processor;
 		}
 
+		public void RegisterPreProcessor(IRemoteControlPreProcessor preprocessor)
```
Comment (Youssef1313) [inline src/Uno.UI.RemoteControl/RemoteControlClient.cs:84]: Do we expect this to be called in a multi-threaded manner?

File: src/Uno.UI.RemoteControl/RemoteControlClient.cs [hunk 3]
```diff
@@ -296,7 +302,20 @@ private async Task ProcessMessages()
...
 						}
 
-						await processor.ProcessFrame(frame);
+						bool skipProcessing = false;
+
+						foreach (var preProcessor in _preprocessors)
+						{
+							if (await preProcessor.SkipProcessingFrame(frame))
+							{
+								skipProcessing = true;
```
Comment (Youssef1313) [inline src/Uno.UI.RemoteControl/RemoteControlClient.cs:311]:
```suggestion
								skipProcessing = true;
								break;
```

File: src/Uno.UI.RemoteControl/RemoteControlClient.cs (hunk 2 above)
Comment (mrlacey) [reply src/Uno.UI.RemoteControl/RemoteControlClient.cs:84]: Hardly expect this to be called at all. I can only think of (have) one, niche scenario where this is necessary.

File: src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.csproj (hunk 1 above)
Comment (mrlacey) [reply src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.csproj:48]: I don't know why they're all there. Just copying what exists when adding something new. 
If this isn't appropriate and all these checks shouldn't be there, I'd like to argue that reviewing all `.csproj` files as a separate task/PR should be undertaken.

File: src/Uno.UI.RemoteControl/RemoteControlClient.cs [hunk 4]
```diff
@@ -38,6 +38,7 @@ public class RemoteControlClient : IRemoteControlClient
 		private readonly (string endpoint, int port)[]? _serverAddresses;
 		private WebSocket? _webSocket;
 		private Dictionary<string, IRemoteControlProcessor> _processors = new Dictionary<string, IRemoteControlProcessor>();
+		private List<IRemoteControlPreProcessor> _preprocessors = new List<IRemoteControlPreProcessor>();
```
Comment (nickrandolph) [inline src/Uno.UI.RemoteControl/RemoteControlClient.cs:41]: Should these be scoped the same way _processors are?

Comment (mrlacey) [reply src/Uno.UI.RemoteControl/RemoteControlClient.cs:41]: The main use case for this is as a way of interacting/intercepting the processing that another processor might do but without having to spread the logic across the two processors (and their assemblies.)

File: src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.csproj (hunk 1 above)
Comment (mrlacey) [reply src/Uno.UI.RemoteControl/Uno.UI.RemoteControl.csproj:48]: Created https://github.com/unoplatform/uno/issues/12409 to tidy up after this PR is merged

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**